File: src/shared/Contracts/Appointment/Responses/RescheduleAppointmentResponse.cs
namespace Contracts.Appointment.Responses;

/// <summary>
/// API response for RescheduleAppointment operation
/// </summary>
/// <param name="AppointmentId">Unique identifier for the rescheduled appointment</param>
/// <param name="NewScheduledDate">New scheduled date and time for the appointment (ISO-8601 format with timezone)</param>
/// <param name="NewDurationMinutes">New duration in minutes for the appointment</param>
/// <param name="UpdatedAt">Date and time when the appointment was rescheduled (ISO-8601 format with timezone)</param>
public record RescheduleAppointmentResponse(
    string AppointmentId,
    DateTimeOffset NewScheduledDate,
    int NewDurationMinutes,
    DateTimeOffset UpdatedAt)
{
    /// <summary>
    /// API Version this response supports
    /// </summary>
    public string ApiVersion => "v1";
}

File: src/client/src/features/appointments/types/RescheduleAppointmentResponse.ts
/**
 * Response contract for reschedule appointment operation
 * Matches the backend RescheduleAppointmentResponse contract
 */
export interface RescheduleAppointmentResponse {
  /**
   * Unique identifier for the rescheduled appointment
   */
  appointmentId: string;

  /**
   * New scheduled date and time for the appointment (ISO-8601 format with timezone)
   */
  newScheduledDate: string;

  /**
   * New duration in minutes for the appointment
   */
  newDurationMinutes: number;

  /**
   * Date and time when the appointment was rescheduled (ISO-8601 format with timezone)
   */
  updatedAt: string;

  /**
   * API Version this response supports
   */
  apiVersion?: string;
}
